assets/src/components/with-featured-image-notice.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * WordPress dependencies
+ */
+import { Notice } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Conditionally displays a notice above the post's 'Featured Image' component.
+ *
+ * If there is no featured image, this displays a notice.
+ * This displays in the sidebar, in the 'Document' panel.
+ *
+ * @param {Function} PostFeaturedImage The featured image component, appearing in the sidebar.
+ * @return {Function} The PostFeaturedImage component, wrapped in a Notice if there's no featured image.
+ */
+export default ( PostFeaturedImage ) => {
+	return withSelect( ( select ) => {
+		return {
+			currentPost: select( 'core/editor' ).getCurrentPost(),
+			editedFeaturedMedia: select( 'core/editor' ).getEditedPostAttribute( 'featured_media' ),
+		};
+	} )( ( ownProps ) => {
+		const { currentPost, editedFeaturedMedia } = ownProps,
+			hasFeaturedMedia = currentPost.featured_media || editedFeaturedMedia;
+
+		function AmpNoticeBlockEdit( props ) {
+			const postFeaturedImage = (
+				<PostFeaturedImage { ...props } />
+			);
+
+			if ( hasFeaturedMedia ) {
+				return postFeaturedImage;
+			}
+
+			return (
+				<Fragment>
+					<Notice status="warning">
+						<span>
+							{ __( 'Featured image is used as poster-portrait-src and is mandatory', 'amp' ) }
```

Comment:
This message should be made more user-friendly, like:

> Selecting a featured image is required.

Reply:
Sure, https://github.com/ampproject/amp-wp/pull/1998/commits/52896509b940921c2c27a445ca45cfb978d86ce0 changes the message:

<img width="670" alt="selecting-featured-image-required" src="https://user-images.githubusercontent.com/4063887/54846002-7dc43600-4ca0-11e9-8150-13925a5f2054.png">